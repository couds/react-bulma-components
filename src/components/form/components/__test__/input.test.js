import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../input';

describe('Input component', () => {
  it('Should exist', () => {
    expect(Input).toMatchSnapshot();
  });
  it('Should have input classname', () => {
    const component = renderer.create(<Input />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Input className="other-class this-is-a-test" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(<Input style={{ height: 250 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be type email and a with success colors', () => {
    const component = renderer.create(<Input color="success" type="email" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be large and readOnly', () => {
    const component = renderer.create(<Input readOnly size="large" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be disabled with placeholder and value', () => {
    const component = renderer.create(
      <Input value="TEST" disabled placeholder="hello tests" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
