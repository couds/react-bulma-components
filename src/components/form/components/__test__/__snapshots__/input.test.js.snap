// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should be disabled with placeholder and value 1`] = `
<input
  className="input"
  disabled={true}
  name={null}
  placeholder="hello tests"
  readOnly={false}
  style={Object {}}
  type="text"
  value="TEST"
/>
`;

exports[`Input component Should be large and readOnly 1`] = `
<input
  className="input is-large"
  disabled={false}
  name={null}
  placeholder=""
  readOnly={true}
  style={Object {}}
  type="text"
  value=""
/>
`;

exports[`Input component Should be type email and a with success colors 1`] = `
<input
  className="input is-success"
  disabled={false}
  name={null}
  placeholder=""
  readOnly={false}
  style={Object {}}
  type="email"
  value=""
/>
`;

exports[`Input component Should concat classname in props with Bulma classname 1`] = `
<input
  className="input other-class this-is-a-test"
  disabled={false}
  name={null}
  placeholder=""
  readOnly={false}
  style={Object {}}
  type="text"
  value=""
/>
`;

exports[`Input component Should exist 1`] = `[Function]`;

exports[`Input component Should have input classname 1`] = `
<input
  className="input"
  disabled={false}
  name={null}
  placeholder=""
  readOnly={false}
  style={Object {}}
  type="text"
  value=""
/>
`;

exports[`Input component Should use inline styles 1`] = `
<input
  className="input"
  disabled={false}
  name={null}
  placeholder=""
  readOnly={false}
  style={
    Object {
      "height": 250,
    }
  }
  type="text"
  value=""
/>
`;
